packages/react-storage/src/components/StorageBrowser/Views/LocationActionView/UploadControls.tsx

```diff
@@ -316,7 +312,27 @@ export const UploadControls = (): JSX.Element => {
 
   return (
     <>
-      {fileSelect}
+      <input
+        data-testid="amplify-file-select"
+        type="file"
+        style={{ display: 'none' }}
+        multiple
+        onChange={({ target }) => {
+          handleFileSelect([...(target.files ?? [])]);
+        }}
+        ref={handleFileInput}
+      />
+      <input
+        data-testid="amplify-folder-select"
+        type="file"
+        style={{ display: 'none' }}
+        onChange={({ target }) => {
+          handleFileSelect([...(target.files ?? [])]);
+        }}
+        // @ts-expect-error webkitdirectory is not typed
+        webkitdirectory=""
+        ref={handleFolderInput}
+      />
```

Comment:
Can we put these directly in the "add" controls so we can remove the need for the effect/additional handling logic?

Reply:
just to make sure I understand the ask, are you saying I make a new component for each input with their respective buttons?

Reply:
Can be addressed later